bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/osgi/ResourceBundleClassLoader.java

```diff
@@ -172,15 +187,26 @@ private Charset getResourceCharset(URL url) {
         return charset;
     }
 
+    /**
+     * The method tests whether the file content can be decoded with the given character set.
+     *
+     * @param f
+     *            name of the file that will be tested
+     * @param charset
+     *            character set to which the file is tested
+     * @return
+     *         passed character set if the file could be decoded with this character set or null if the decode fails
```

Comment:
Might it make sense to refactor this method to simply do the check and return a boolean? Would make much more sense to me...